src/plugins/data/public/ui/dataset_selector/dataset_selector.tsx

```diff
@@ -57,15 +58,26 @@ export const DatasetSelector = ({
 
     if (!isMounted.current) return;
 
+    const recentDatasets = datasetService.getRecentDatasets();
     const fetchedDatasets =
       fetchedIndexPatternDataStructures.children?.map((pattern) =>
         typeConfig.toDataset([pattern])
       ) ?? [];
-    setDatasets(fetchedDatasets);
+    const filteredFetchedDatasets = fetchedDatasets.filter(
+      (dataset) => !recentDatasets.some((ds) => ds.id === dataset.id)
+    );
+    const allDatasets = [
+      ...recentDatasets.map((ds) => ({ dataset: ds, recent: true })),
```

Comment:
would there be invalid/outdated datasets? should we check and make sure every recent dataset appears in the fetchedDatasets?

Reply:
No, because fetched datasets only contain index patterns